### PR TITLE
don't fail silently when setting config variable that doesn't exist

### PR DESCRIPTION
**Pull request by wakamex**
* implement `disable_new_attribs` function of freezable classes
* allow `frozen` and `no_new_attribs` to be enabled by default through constructor parameters

### CODE REVIEW

File: src/elfpy/types.py [hunk 1]
```diff
@@ -29,21 +29,34 @@ def freezable(cls: Type) -> Type:
 
     @wraps(cls, updated=())
     class FrozenClass(cls):
-        """Subclass cls to add frozen check & freeze function"""
+        """Subclass cls to add frozen check & freeze function
+
+        TODO: resolve why pylint throws a no-member error on freezable wrapped
```
Comment (dpaiton) [inline src/elfpy/types.py:34]: Can you change this to the sphynx todo format:
`.. todo:: `

File: tests/test_freezable.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+from elfpy.types import freezable
+
+
+@freezable
+@dataclass
+class FreezableClass:
+    """Freezable class for testing"""
+
+    def __init__(self):
```
Comment (dpaiton) [inline tests/test_freezable.py:15]: No need to define init on a dataclass. In fact I'm surprised it let you at all. Just set it as a member attribute:

```
@freezable
@dataclass
class FreezableClass:
    """Freezable class for testing"""
    
    existing_attrib: int = 1
```

**Review by dpaiton — APPROVED** [review]
Looks great! Thanks a lot for cleaning up the test as well. Please fix up the couple of small comments & then it's good to go.

File: src/elfpy/types.py [hunk 3]
```diff
@@ -24,40 +24,43 @@ def to_description(description: str) -> dict[str, str]:
...
-            if hasattr(self, attrib) and hasattr(self, "frozen") and getattr(self, "frozen"):
-                raise AttributeError(f"{self.__class__.__name__} is frozen, cannot change attribute '{attrib}'.")
-            if not hasattr(self, attrib) and hasattr(self, "no_new_attribs") and getattr(self, "no_new_attribs"):
-                raise AttributeError(
-                    f"{self.__class__.__name__} has no_new_attribs set, cannot add attribute '{attrib}'."
-                )
-            super().__setattr__(attrib, value)
+            def __init__(self, *args, frozen=frozen, no_new_attribs=no_new_attribs, **kwargs) -> None:
+                super().__init__(*args, **kwargs)
+                print(f"FreezableClass.__init__(class={cls.__name__},{frozen=}, {no_new_attribs=})")
```
Comment (sentilesdal) [inline src/elfpy/types.py:40]: remove print

File: tests/test_freezable.py [hunk 4]
```diff
@@ -70,3 +113,22 @@ def test_freezable_when_frozen_is_set_and_no_new_attribs_is_set(self):
             freezable_object.existing_attrib = 2
         with self.assertRaises(AttributeError):
             freezable_object.new_attrib = 1  # pylint: disable=attribute-defined-outside-init # type: ignore
+
+    def test_freezable_when_frozen_is_set_and_no_new_attribs_is_set_by_default(self):
+        """
+        freezable object when not frozen is NOT set, but no_new_attribs IS set
```
Comment (sentilesdal) [inline tests/test_freezable.py:119]: update this comment, its copy pasta

File: tests/test_freezable.py [hunk 5]
```diff
@@ -70,3 +113,22 @@ def test_freezable_when_frozen_is_set_and_no_new_attribs_is_set(self):
...
+        desired behavior is: CAN change attributes, but can NOT add attributes
+        """
+        freezable_object = FreezableClassFrozenNoNewAttribsByDefault()
+        with self.assertRaises(AttributeError):
+            freezable_object.existing_attrib = 2
+        with self.assertRaises(AttributeError):
+            freezable_object.new_attrib = 1  # pylint: disable=attribute-defined-outside-init # type: ignore
+
+    def test_config_cant_add_new_attribs(self):
```
Comment (sentilesdal) [inline tests/test_freezable.py:128]: this should be in a test_config file, not test_freezable

**Review by sentilesdal — APPROVED** [review]
looks good after nits fixed.  really nice improvement.

**Pull request merged**